### ISSUE TITLE
Alert rules: "Unauthorized" warning when accessing as anonymous user

### ISSUE THREAD

**Issue by Birkman** [grafana/grafana#83033]
### What happened?

After I enabled anonymous authentication and accessed Alerting --> Alert rules, I got "Unauthorized" warning:
![image](https://github.com/grafana/grafana/assets/8317316/7e8f931e-bec5-497a-843e-26bbb2cfbc18)

### What did you expect to happen?

Behavior should be identical to a regular user who is a viewer, i.e.: no "Unauthorized" warning.
BTW, I see the same warning when anonymous has role of Editor.   

### Did this work before?

Unknown

### How do we reproduce it?

1. Run: docker run -d -p 3000:3000 --name=grafana -e "GF_AUTH_ANONYMOUS_ENABLED=true" grafana/grafana
2. Go to: Alerting --> Alert rules
3.

### Environment (with versions)?

Grafana: Grafana v10.3.1 (00a22ff8b2)
OS: macOS 13.6.3 (22G436)
Browser: Chrome 121.0.6167.184 (Official Build) (x86_64)

### Grafana platform?

Docker

**Comment by tonypowa** [comment]
hi @Birkman 

Thank you for creating this issue

I don't see Alerting listed as something that users with Viewer, and Editor role can access
We might need to change the doc
https://grafana.com/docs/grafana/latest/administration/roles-and-permissions/#organization-roles

I will review this internally and clarify which roles allow editing Alerting resources, and make any changes to the docs if necessary

**Comment by ThomasRasmussen** [comment]
Can confirm it to be an issue - also in the lastest 11.0.0 release. It doesn't seem to impact any functionality.

**Comment by chenjie4255** [comment]
> Can confirm it to be an issue - also in the lastest 11.0.0 release. It doesn't seem to impact any functionality.

If user deploys Grafana behind Nginx and enables Basic Auth in Nginx, this may trigger a basic auth login dialog repeatedly.

**Comment by amartincolby** [comment]
I know that "same" comments aren't helpful, but they prevent something from being marked "stale." This is not stale.

**Comment by Konubinix** [comment]
I agree, this is an issue I just encountered (with version 10.0.1). It's very annoying.

**Comment by ranvit** [comment]
Sorry if I've misunderstood this thread, but I'm trying to debug something similar.

I'm on v11.1 and I have 2 instances of grafana.

My test instance has `GF_AUTH_ANONYMOUS_ORG_ROLE=Editor, GF_USERS_EDITORS_CAN_ADMIN=true` and on this instance I'm able to view an alert rule (aka go to `mygrafana.io/alerting/grafana/<alert-dashboard-uid>/view`)

But on my live instance that has `GF_AUTH_ANONYMOUS_ORG_ROLE=Viewer, GF_USERS_VIEWERS_CAN_EDIT=true`, I'm unable to view the alert rules folder and I'm getting `Permissions needed: all of alert.rules:read, folders:read`
These settings are sufficient for viewing and testing out edits on dashboards, without allowing users to save those changes to the live instance. But it seems like alerts are completely closed off to the Viewer role :/

**Comment by wjordan** [comment]
Here is the code that triggers this error alert, a call to `/api/user` to prevent analytics tracking from being applied to new users, added in #61005:
https://github.com/grafana/grafana/blob/ce0d48746f85448478031f840b0e247d6631b816/public/app/features/alerting/unified/Analytics.ts#L157
